internal/agentdeployer/factory.go

```diff
@@ -80,23 +67,51 @@ func Factory(options FactoryOptions) (AgentDeployer, error) {
 		// FIXME: this docker-compose scenario contains both agent and service
 		return nil, nil
 	case "k8s":
-		if _, err := os.Stat(agentDeployerPath); err == nil {
-			opts := KubernetesAgentDeployerOptions{
-				Profile:        options.Profile,
-				DefinitionsDir: agentDeployerPath,
-				StackVersion:   options.StackVersion,
-				PolicyName:     options.PolicyName,
-				DataStream:     options.DataStream,
-				RunSetup:       options.RunSetup,
-				RunTestsOnly:   options.RunTestsOnly,
-				RunTearDown:    options.RunTearDown,
-			}
-			return NewKubernetesAgentDeployer(opts)
+		opts := KubernetesAgentDeployerOptions{
+			Profile:        options.Profile,
+			DefinitionsDir: agentDeployerPath,
+			StackVersion:   options.StackVersion,
+			PolicyName:     options.PolicyName,
+			DataStream:     options.DataStream,
+			RunSetup:       options.RunSetup,
+			RunTestsOnly:   options.RunTestsOnly,
+			RunTearDown:    options.RunTearDown,
 		}
+		return NewKubernetesAgentDeployer(opts)
 	}
 	return nil, fmt.Errorf("unsupported agent deployer (name: %s)", agentDeployerName)
 }
 
+func selectAgentDeployerType(options FactoryOptions) (string, string, error) {
+	devDeployPath, err := FindDevDeployPath(options)
+	if errors.Is(err, os.ErrNotExist) {
+		return "default", "", nil
+	}
+	if err != nil {
+		return "", "", fmt.Errorf("can't find \"%s\" directory: %w", options.DevDeployDir, err)
+	}
+
+	agentDeployerName, err := findAgentDeployer(devDeployPath)
+	if err != nil {
+		logger.Debugf("Not found any agent deployer, using default one")
+		return "default", "", nil
+	}
```

Comment:
The error returned by `findAgentDeployer` could be that there are more than one deployer folders defined.

It needs to be taken into account that too. One option could be adding an specific error an check if that error is returned.